pkg/storage/unified/resource/authz.go

```diff
@@ -0,0 +1,115 @@
+package resource
+
+import (
+	context "context"
+	"net/http"
+
+	"github.com/grafana/grafana/pkg/apimachinery/identity"
+)
+
+// Quickly check if an resource (in a folder) should be seen
+// returns true when the item should be excluded from list results
+type ResourceReadFilter func(namespace, name, folder string) bool
+
+// Authorizer checks if a user is able to access a resource
+// Functions return an ErrorResult when not accessible so that access details
+// can be better communicated in the result
+type Authorizer interface {
+	CanCreate(ctx context.Context, id identity.Requester, key *ResourceKey) *ErrorResult
```

Comment:
I'm not wild about how the `ErrorResult` looks -- but it allows more information that a boolean response.

Myabe these should be `AssertCreate()`?  any better ideas?